### ISSUE TITLE
What if my .exe has two services in it?

### ISSUE THREAD

**Issue by tsahi** [iswix-llc/iswix-tutorials#3]
I created a Windows Service project, where the assembly contains two services, i.e. two objects are added to the `ServiceBase` array. How do you support that with the IsWix designer?

**Comment by chrpai** [comment]
I've never heard of or tried to do multiple services per EXE assembly.  I have done a generic service host pattern where the service/EXE spins up a bunch of background workers (injected by reflection).

IsWiX/WiX/MSI does support creating two services pointing to the same key file (.exe).  I don't know enough about service base though to know how you would tell each service how to behave differently.  Maybe you could put something in the Parameters registry subkey of each service and that would influence it.   I have done this with things like servany.exe and nssm.exe.

**Comment by chrpai** [comment]
Take a look at: https://stackoverflow.com/questions/9116870/wix-how-to-install-multiple-windows-services-with-one-executable

**Comment by tsahi** [comment]
I found IsWiX when I looked for a way to maintain the list of files to copy to the target machine in a reasonable way. eventually I just used the IsWix v3 Merge Module template, because I already had all the code for the rest of the installer logic.

**Comment by chrpai** [comment]
Merge Module authoring was actually the v1.0 story for IsWiX.   In the beginning it was about giving developers a way to update buckets of files consumed by other tools (namely InstallShield).   I then changed jobs and enhanced the story to be able to create installers and bootstrappers using only WiX. 

I wrote these blog articled 8-10  years ago:

http://blog.deploymentengineering.com/2009/06/installshield-2010-vs-wix-30.html
http://blog.iswix.com/2010/05/distributed-setup-development-revisited.html
http://blog.iswix.com/2011/03/installation-collaboration-workflows.html
